features/timeseries/src/main/java/org/opennms/netmgt/timeseries/TimeseriesStorageManagerImpl.java

```diff
@@ -53,7 +53,7 @@ public class TimeseriesStorageManagerImpl implements TimeseriesStorageManager {
 
     public TimeseriesStorageManagerImpl() {
         this(new ServiceLookupBuilder(new ServiceRegistryLookup(DefaultServiceRegistry.INSTANCE))
-                .blocking()
+                .blocking(ServiceLookupBuilder.GRACE_PERIOD_MS, ServiceLookupBuilder.LOOKUP_DELAY_MS, 0)
```

Comment:
If this is equivalent to 0 ms wait,  simply remove blocking

Reply:
There are *two* independent periods that are checked and `BlockingServiceLookup` won't give up until *both* expire. We are only changing the second one, wait period, to zero:
1. grace period -- the first argument to `ServiceLookupBuilder.blocking`, `gracePeriodInMs`.
2. wait period -- the third argument to `ServiceLookupBuilder.blocking`, `waitTimeMs`.

(the second argument to `ServiceLookupBuilder.blocking` is `sleepTimeInMs`, which is the amount of time to wait between checks to see if the service is available)

```
        // A service matching the filter is not currently available.
        // Wait until the system has finished starting up (uptime >= grace period)
        // while ensuring we've waited for at least WAIT_PERIOD_MS before aborting the search.
        final long waitUntil = System.currentTimeMillis() + this.waitTimeMs;
        while (uptimeSupplier.get() < this.gracePeriodInMs
                || System.currentTimeMillis() < waitUntil) {
            try {
                Thread.sleep(this.lookupDelayMs);
...
```
-- https://github.com/OpenNMS/opennms/blob/develop/core/soa/src/main/java/org/opennms/core/soa/lookup/BlockingServiceLookup.java#L65

Reply:
got it. Makes sense!